homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,68 @@ def state(self):
-                return
+        # Some state data needs to be rounded to whole values or converted to
+        # percentages
+        if self.type in ['precip_probability', 'cloud_cover', 'humidity']:
+            state = state * 100
+        if (self.type in ['precip_probability', 'dew_point', 'temperature',
+                          'apparent_temperature', 'cloud_cover', 'humidity',
+                          'pressure', 'ozone']):
+            state = round(state, 1)
 
-            elif self.type == 'daily_summary':
-                self.forecast_client.update_daily()
-                self._state = self.forecast_client.data_daily.summary
-                return
+        return state
 
-        except forecastio.utils.PropertyUnavailable:
-            return
 
-        self.forecast_client.update_currently()
-        data = self.forecast_client.data_currently
+def convert_to_camel(data):
+    """
+    Convert snake case (foo_bar_bat) to camel case (fooBarBat).
 
-        try:
-            if self.type == 'summary':
-                self._state = data.summary
-            elif self.type == 'icon':
-                self._state = data.icon
-            elif self.type == 'nearest_storm_distance':
-                self._state = data.nearestStormDistance
-            elif self.type == 'nearest_storm_bearing':
-                self._state = data.nearestStormBearing
-            elif self.type == 'precip_intensity':
-                self._state = data.precipIntensity
-            elif self.type == 'precip_type':
-                self._state = data.precipType
-            elif self.type == 'precip_probability':
-                self._state = round(data.precipProbability * 100, 1)
-            elif self.type == 'dew_point':
-                self._state = round(data.dewPoint, 1)
-            elif self.type == 'temperature':
-                self._state = round(data.temperature, 1)
-            elif self.type == 'apparent_temperature':
-                self._state = round(data.apparentTemperature, 1)
-            elif self.type == 'wind_speed':
-                self._state = data.windSpeed
-            elif self.type == 'wind_bearing':
-                self._state = data.windBearing
-            elif self.type == 'cloud_cover':
-                self._state = round(data.cloudCover * 100, 1)
-            elif self.type == 'humidity':
-                self._state = round(data.humidity * 100, 1)
-            elif self.type == 'pressure':
-                self._state = round(data.pressure, 1)
-            elif self.type == 'visibility':
-                self._state = data.visibility
-            elif self.type == 'ozone':
-                self._state = round(data.ozone, 1)
-
-        except forecastio.utils.PropertyUnavailable:
-            pass
+    This is not pythonic, but needed for certain situations
+    """
+    components = data.split('_')
+    return components[0] + "".join(x.title() for x in components[1:])
```

Comment:
@fabaff You mentioned that we do this `if/elif/elif/elif...` in other modules.
If you guys like the looks of this, we could pull it out into a utility/helper and use it in those modules.